src/python/pants/core/goals/repl.py

```diff
@@ -79,17 +81,26 @@ class ReplRequest:
     digest: Digest
     args: Tuple[str, ...]
     extra_env: FrozenDict[str, str]
+    use_nailgun: Digest
+    append_only_caches: FrozenDict[str, str]
+    run_in_workspace: bool
 
     def __init__(
         self,
         *,
         digest: Digest,
         args: Iterable[str],
         extra_env: Optional[Mapping[str, str]] = None,
+        use_nailgun: Digest = EMPTY_DIGEST,
```

Comment:
I don't think that using nailgun (as implemented currently) will work in the workspace. The server runs in a long lived temporary directory, and we acquire exclusive access to a server. I also don't think that nailgun will work for a `repl`, since they usually require TTY access, which we have gone to great lengths to support in `nails`, but which a JVM server will not support.

Reply:
The PR ended up not using nailgun although I had threaded it through. Removed.